src/SeqCli/Cli/Commands/Bench/BenchCommand.cs

```diff
@@ -0,0 +1,213 @@
+            a => _reportingServerApiKey = a);
+    }
+    
+    protected override async Task<int> Run()
+    {
+        try
+        {
+            var connection = _connectionFactory.Connect(_connection);
+            _reportingLogger = BuildReportingLogger();
+            var cases = ReadCases(_cases);
+            var runId = Guid.NewGuid().ToString("N").Substring(0, 4);
+            var start = _range.Start ?? DateTime.UtcNow.AddDays(-7);
+            var end = _range.End;
+
+            foreach (var c in cases.Cases)
+            {
+                var timings = new BenchCaseTimings();
+                object? lastResult = null;
+                
+                foreach (var i in Enumerable.Range(1, _runs))
+                {
+                    var response = await connection.Data.QueryAsync(
+                        c.Query,
+                        _range.Start ?? DateTime.UtcNow.AddDays(-7),
+                        _range.End,
+                        SignalExpressionPart.FromIntersectedIds(c.Signals)
+                    );
+
+                    timings.PushElapsed(response.Statistics.ElapsedMilliseconds);
+                    var isScalarResult = response.Rows.Length == 1 && response.Rows[0].Length == 1;
+
+                    if (i == _runs && isScalarResult)
+                    {
+                        lastResult = response.Rows[0][0];
+                    }
+                }
+                
+                using (lastResult != null ? LogContext.PushProperty("LastResult", lastResult) : null)
+                using (LogContext.PushProperty("MinElapsed", timings.MinElapsed))
+                using (LogContext.PushProperty("MaxElapsed", timings.MaxElapsed))
+                using (LogContext.PushProperty("Runs", _runs))
+                using (LogContext.PushProperty("Signals", c.Signals))
+                using (LogContext.PushProperty("Start", start))
+                using (LogContext.PushProperty("StandardDeviationElapsed", timings.StandardDeviationElapsed))
+                using (end != null ? LogContext.PushProperty("End", end) : null)
+                using (LogContext.PushProperty("Query", c.Query))
+                {
+                    _reportingLogger.Information(
+                        "Bench run {BenchRunId} for query {Id}. Mean {MeanElapsed:N0} ms with relative dispersion {RelativeStandardDeviationElapsed:N2}", 
+                        runId, c.Id,
+                        timings.MeanElapsed, 
+                        timings.RelativeStandardDeviationElapsed);
+                }
+            }
+
+            return 0;
+        } catch (Exception ex)
+        {
+            Log.Error(ex, "Benchmarking failed: {ErrorMessage}", ex.Message);
+            return 1;
+        }
+    }
+
+    /// <summary>
+    /// Build a second Serilog logger for logging benchmark results. 
+    /// </summary>
+    Logger BuildReportingLogger()
+    {
+        return string.IsNullOrWhiteSpace(_reportingServerUrl) 
```

Comment:
Logger configuration could be saved to a temporary and mutated - e.g.:

```csharp
var loggerConfiguration = new LoggerConfiguration()
    .Enrich.FromLogContext()
    .WriteTo.Console();

if (!string.IsNullOrWhiteSpace(_reportingServerUrl))
    loggerConfiguration.WriteTo.Seq(
                    _reportingServerUrl, 
                    apiKey: string.IsNullOrWhiteSpace(_reportingServerApiKey) ? null : _reportingServerApiKey,
                    period: TimeSpan.FromMilliseconds(1));

return loggerConfiguration.CreateLogger();
```

Not any more succinct unfortunately - but might be slightly preferable without the duplication?